src/lib/providers/remoteFsProvider.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  TreeDataProvider,
+  TreeItem,
+  Uri,
+  TreeItemCollapsibleState,
+  ThemeIcon,
+  EventEmitter,
+  Event,
+} from 'vscode';
+import { FileLink, RemoteFsDirectory } from '../types';
+
+const {
+  getDirectoryContentsByPath,
+} = require('@hubspot/cli-lib/api/fileMapper');
+const { getPortalId } = require('@hubspot/cli-lib');
+const { FOLDER_DOT_EXTENSIONS } = require('@hubspot/cli-lib/lib/constants');
+
+function isPathFolder(path: string) {
+  const splitPath = path.split('/');
+  const fileOrFolderName = splitPath[splitPath.length - 1];
+  const splitName = fileOrFolderName.split('.');
+
+  return !(
+    splitName.length > 1 && FOLDER_DOT_EXTENSIONS.indexOf(splitName[1]) === -1
+  );
+}
```

Comment:
The file endpoint tells us whether a given path is a folder or not, but only after we fetch its contents. We render it in the tree when it shows up as a child first, so we have to infer from its name whether it's a file or folder. For example our initial root call returns something like
```
{
  "id": "/",
  "name": "",
  "folder": true, <-- tells us that the root '/' is a folder
  "children": [
    "test_tabs.module", <-- we want to render these in the tree now, but won't get the above info until this gets expanded
    "Marketplace",
    "boilerplate-test",
    "growth-test",
    "practicum",
    "local-CmsDefaultSystemPages"
  ],
  "updatedAt": 1642781510812,
  "createdAt": 1642781510812,
  "archivedAt": null
}
```